pkg/rabbit/service.go

```diff
@@ -203,7 +204,35 @@ func isReady(conditions []v1beta1.Condition) bool {
 	return numConditions == 0
 }
 
-func (r *Rabbit) RabbitMQURL(ctx context.Context, clusterRef *rabbitv1beta1.RabbitmqClusterReference) (*url.URL, error) {
+func (r *Rabbit) RabbitMQURL(ctx context.Context, clusterRef *rabbitv1beta1.RabbitmqClusterReference, parentNamespace string) (*url.URL, error) {
+	if clusterRef.ConnectionSecret != nil {
+		ns := "default"
+		if clusterRef.Namespace != "" {
+			ns = clusterRef.Namespace
+		} else if parentNamespace != "" {
```

Comment:
parentNamespace seems to be broker.namespace. Would that ever be empty? 

I get why we're defaulting here (because the rabbitmqClusterReference struct has that description), but I'm wondering if this would be intuitive enough for a user. Having it default to the Broker's namespace would need to be documented and there's no guarantee a user would actually see it. I think having namespace be required or just having a simple `+kubebuilder:default:=default` if we want to leave it optional would be better.